apps/Standalone/src/designer/app/AzureLogicAppsDesigner/Utilities/Workflow.ts

```diff
@@ -77,7 +77,7 @@ export class WorkflowUtility {
       for (const settingName of Object.keys(appsettings)) {
         const settingValue = appsettings[settingName] !== undefined ? appsettings[settingName] : '';
         // Don't replace if the setting value is a KeyVault reference
-        if (settingValue.startsWith('@Microsoft.KeyVault(')) {
+        if (typeof settingValue === 'string' && settingValue.startsWith('@Microsoft.KeyVault(')) {
```

Comment:
instead of typeof, you could also do this
```suggestion
        if (settingValue.startsWith && settingValue.startsWith('@Microsoft.KeyVault(')) {
```

Reply:
yeah I actually am thinking instead of resolving these non-string app settings, thinking we just not resolve them altogether because it might run into an error later on in the replace